src/main/java/ee/tuleva/onboarding/auth/ocsp/OCSPService.java

```diff
@@ -0,0 +1,125 @@
+package ee.tuleva.onboarding.auth.ocsp;
+
+import static ee.tuleva.onboarding.auth.exception.AuthenticationException.Code.UNABLE_TO_TEST_USER_CERTIFICATE;
+import static ee.tuleva.onboarding.auth.ocsp.OCSPResponseType.EXPIRED;
+import static ee.tuleva.onboarding.auth.ocsp.OCSPResponseType.GOOD;
+import static ee.tuleva.onboarding.auth.ocsp.OCSPResponseType.REVOKED;
+import static ee.tuleva.onboarding.auth.ocsp.OCSPResponseType.UNKNOWN;
+import static org.bouncycastle.cert.ocsp.OCSPResp.MALFORMED_REQUEST;
+import static org.bouncycastle.cert.ocsp.OCSPResp.SUCCESSFUL;
+import static org.bouncycastle.cert.ocsp.OCSPResp.UNAUTHORIZED;
+import static org.springframework.http.HttpMethod.POST;
+
+import ee.tuleva.onboarding.auth.exception.AuthenticationException;
+import ee.tuleva.onboarding.auth.exception.AuthenticationException.Code;
+import java.io.IOException;
+import java.security.cert.X509Certificate;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Base64;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.cert.ocsp.BasicOCSPResp;
+import org.bouncycastle.cert.ocsp.OCSPException;
+import org.bouncycastle.cert.ocsp.OCSPResp;
+import org.bouncycastle.cert.ocsp.RevokedStatus;
+import org.bouncycastle.cert.ocsp.SingleResp;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.RestTemplate;
+
+@Slf4j
+@Service
+@AllArgsConstructor
+public class OCSPService {
+  private final RestTemplate restTemplate;
+  public static final String BEGIN_CERT = "-----BEGIN CERTIFICATE-----";
+  public static final String END_CERT = "-----END CERTIFICATE-----";
+  public static final String LINE_SEPARATOR = System.getProperty("line.separator");
+
+  public OCSPResponseType checkCertificate(OCSPRequest request) {
+    X509Certificate certificate = request.getCertificate();
+    if (hasCertificateExpired(certificate)) {
+      return EXPIRED;
+    }
+    return checkCertificateStatus(request);
+  }
+
+  private boolean hasCertificateExpired(X509Certificate certificate) {
+    Clock clock = Clock.systemDefaultZone();
```

Comment:
This is not better than using System.currentTimeMillis. You can still not mock it. Dependency injection is the keyword here. Create a Clock bean and inject it.

Reply:
Take a look at how it's injected and mocked in: https://github.com/TulevaEE/onboarding-service/blob/master/src/test/groovy/ee/tuleva/onboarding/conversion/UserConversionServiceSpec.groovy#L35